bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConfiguration.java

```diff
@@ -115,14 +138,15 @@ public CloudBigtableConfiguration build() {
   public CloudBigtableConfiguration(String projectId, String zoneId, String clusterId,
       Map<String, String> additionalConfiguration) {
     this.configuration = new HashMap<>(additionalConfiguration);
-    setValue(BigtableOptionsFactory.PROJECT_ID_KEY, projectId);
-    setValue(BigtableOptionsFactory.ZONE_KEY, zoneId);
-    setValue(BigtableOptionsFactory.CLUSTER_KEY, clusterId);
+    setValue(BigtableOptionsFactory.PROJECT_ID_KEY, projectId, "Project ID");
+    setValue(BigtableOptionsFactory.ZONE_KEY, zoneId, "Zone ID");
+    setValue(BigtableOptionsFactory.CLUSTER_KEY, clusterId, "Cluster ID");
   }
 
-  private void setValue(String key, String value) {
+  private void setValue(String key, String value, String type) {
     Preconditions.checkArgument(!configuration.containsKey(key),
       String.format("%s was set twice", key));
+    Preconditions.checkArgument(value != null, String.format("%s must be set.", type));
```

Comment:
Type is more understandable from a user perspective in this case.  They have to set the projectId, zoneId or clusterId via a -- dataflow parameter (i.e. --projectId).  The key is only pertinent if a user is a more advanced user and sets a withConfiguration(key, value).  A user of the basic setup wouldn't know what a "google.bigtable.project.id" is, but would understand "Project ID."